### PR TITLE
Add crystvis features #12

### LINKED ISSUE #12: Integrate further crystvis-js options
We need to add the various options available in crystvis-js to the visualiser sidebar in Galaxy.

### PR DESCRIPTION
**Pull request by patrick-austin**
Closes #12 
This adds support for relevant crystvis features through the Galaxy UI.
image

Note that the export screenshot will only output a blank image.
**EDIT:** Having looked into this some more, integrating with the existing Galaxy screenshot buttons would require a change to Crystvis, to allow us to provide generate a renderer with `WebGLRenderer({preserveDrawingBuffer: true})`. There may be some further tinkering required in Galaxy, but this would be the first step.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Login
  2. Open visualisations
  3. Select crystvis, and chose a cell or cif file
  4. Test features work as expected, and labels make sense

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: config/plugins/visualizations/crystvis/config/crystvis.xml [hunk 1]
```diff
@@ -7,33 +7,266 @@
...
+            <name>supercell</name>
+            <type>conditional</type>
+            <test_param>
+                <name>enabled</name>
+                <label>Supercell</label>
+                <type>boolean</type>
+                <optional>true</optional>
+                <help>Display as a supercell, with the size in each dimension given as an integer number of lattice vectors</help>
+            </test_param>
+            <cases>
```
Comment (joelvdavies) [inline config/plugins/visualizations/crystvis/config/crystvis.xml:59]: I assume this is correct but just wondered why there is a \<cases> within another \<cases> tag, and similarly with the \<inputs>.

File: config/plugins/visualizations/crystvis/config/crystvis.xml [hunk 2]
```diff
@@ -7,33 +7,266 @@
...
+            <label>van der Waals Radii Scale</label>
+            <name>vdwScaling</name>
+            <help>Rescale van der Waals radii by a constant factor</help>
+            <type>float</type>
+            <value>1.0</value>
+        </input>
+        <input>
+            <name>supercell</name>
+            <type>conditional</type>
+            <test_param>
```
Comment (joelvdavies) [inline config/plugins/visualizations/crystvis/config/crystvis.xml:52]: More of a question about muon galaxy but what are these test_params's for, it seems to be what is actually displayed but if that's the case what are the two lines above defining?

File: config/plugins/visualizations/crystvis/config/crystvis.xml [hunk 3]
```diff
@@ -7,33 +7,266 @@
...
+                            <name>box_start</name>
+                            <type>text</type>
+                            <value>0 0 0</value>
+                        </inputs>
+                        <inputs>
+                            <label>Box End</label>
+                            <name>box_end</name>
+                            <type>text</type>
+                            <value>1 1 1</value>
+                            <help>All atoms with a co-ordinates between the start and end values will be returned, co-ordinates should be space separated</help>
```
Comment (joelvdavies) [inline config/plugins/visualizations/crystvis/config/crystvis.xml]: Minor typo - could do without the 'a'. Also perhaps 'returned' should be 'displayed' in this context?

File: config/plugins/visualizations/crystvis/config/crystvis.xml [hunk 4]
```diff
@@ -7,33 +7,266 @@
...
+                            <type>text</type>
+                            <value>0 0 0</value>
+                        </inputs>
+                        <inputs>
+                            <label>Sphere Radius</label>
+                            <name>sphere_radius</name>
+                            <type>float</type>
+                            <value>1.0</value>
+                            <min>0</min>
+                            <help>All atoms within radius of the start value will be returned, co-ordinates should be space separated</help>
```
Comment (joelvdavies) [inline config/plugins/visualizations/crystvis/config/crystvis.xml]: Same comment about returned.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Just a minor typo, a suggestion and some questions about the format in general. I also assume the tests that are failing are okay?

**Review by joelvdavies — APPROVED** [review]
Looks good to me.

**Comment by patrick-austin** [comment]
Regarding the failing tests: after syncing with the latest upstream code, [Mulled Unit Tests / Test (3.7)](https://github.com/galaxyproject/galaxy/actions/runs/3647908944/jobs/6160693185), [Test Galaxy packages / Test (3.7)](https://github.com/galaxyproject/galaxy/actions/runs/3647908932/jobs/6160692693), and [Test Galaxy packages / Test (3.10)](https://github.com/galaxyproject/galaxy/actions/runs/3647908932/jobs/6160692810) are all failing upstream in the 22.05 release branch. The cause of `Build docs / build (3.7)` failure is documented in #27

**Pull request merged**